src/pymodaq/daq_utils/managers/modules_manager.py

```diff
@@ -192,24 +225,43 @@ def parameter_tree_changed(self, param, changes):
                 pass
 
     def get_det_data_list(self):
-        self.connect_detectors()
+        """Construct the list of data channels that are available with the current configuration of the module manager.
+
+        It will display all the data channels as selectable parameters in the module manager UI.
 
-        datas = self.grab_datas()
+        This method is called by the "Probe detectors’ data" button of the module manager UI.
 
+        """
         data_list0D = []
         data_list1D = []
         data_list2D = []
         data_listND = []
 
-        for k in datas.keys():
-            if 'data0D' in datas[k].keys():
-                data_list0D.extend([f'{k}/{name}' for name in datas[k]['data0D'].keys()])
-            if 'data1D' in datas[k].keys():
-                data_list1D.extend([f'{k}/{name}' for name in datas[k]['data1D'].keys()])
-            if 'data2D' in datas[k].keys():
-                data_list2D.extend([f'{k}/{name}' for name in datas[k]['data2D'].keys()])
-            if 'dataND' in datas[k].keys():
-                data_listND.extend([f'{k}/{name}' for name in datas[k]['dataND'].keys()])
+        for detector in self.detectors:
+            detector_acquisition = detector.data_to_save_export
+            det_name = detector.detector_name
+
+            try:
+                # If the detector is not grabbing (detector_acquisition is None), we send a grab signal. But if it is
+                # grabbing, we do not need to send the grab signal, which would freeze the acquisition.
+                if detector_acquisition is None:
+                    detector.grab_done_signal.connect(self.det_done)
+                    detector_acquisition = self.grab_one_detector(detector)
+                    detector.grab_done_signal.disconnect(self.det_done)
```

Comment:
> this method will not work generally. It will only work when no grab has been done yet on the detector. However if you did a first grab, then stops, then look at this attribute, its value will not be none...

Yes I think that is the behavior we need.

If we are in a state of a dashboard with two detectors, with the first one acquiring and the second one has never acquired (buffer = None). If we launch a snap to every detectors, then we will freeze the acquisition of the first one, which is not what we want. The intention here is to send a grab signal only when needed, so only to detectors with empty buffer, which should not be of much consequence since it means they are just "off". That is why the grab_one_detector has been added.

I try to explained it better in the docstrings but changed nothing in the logic: 8149a13607dc38f31a814fbe92f5af2987c5bee6

I don’t think I understand your last question. Maybe there is a better way to get this information without grabbing?

Reply:
Moreover one could add another ROI between the last grab and the next, so the buffer does not reflects the real data content

Reply:
So still not convinced about doing the modification

Reply:
I got back to the original get_det_data_list and remove the grab_one_detector method.
But I am quite convinced we will come back to them at some point ;)

Reply:
Here: ffda23ed17300df9b83a2ce517eb3470368c0550.